File: test/cv2.jl
module cv2

export cv2

function imread(filename::String)
    hnd = ccall((:imread,"./libcv2"),Ptr{Void},(Ptr{Uint8},), bytestring(filename))
    println(hnd)
    return hnd
end

function imshow(winName::String, mat::Ptr{Void})
    println(mat)
    ccall((:imshow, "./libcv2"), Void, (Ptr{Uint8}, Ptr{Void}), bytestring(winName), mat)
end

function waitKey(delay::Int64)
    ccall((:waitKey, "./libcv2"), Void, (Int64,), delay)
end
end

img = cv2.imread("/home/sunny/workspace/Earth_Mars_Comparison.jpg")
cv2.imshow("test", img)
cv2.waitKey(0)

File: src/cpp/cv2.cpp
#include "cv2.h"
// include iostream header file for test
#include <iostream>

using namespace std;

cv::Mat* imread(char *filename)
{
    cv::Mat img = cv::imread(filename);
    cv::Mat *image = new Mat();
    img.copyTo(*image);

	return image;
}

void imshow(char *winName, Mat* mat)
{
	if (!mat->empty())
	{
		cv::imshow(winName, *mat);
	}
	else
	{
		cout << "Empty image" << endl;
	}
}

void waitKey(int delay)
{
    cv::waitKey(delay);
}


